### PR TITLE
[ARCTIC-1273] When using derby to execute quickstart, the table was erroneously accessed before it was created

### LINKED ISSUE #1273: [Bug]: When using derby to execute quickstart, the table was erroneously accessed before it was created
### What happened?

When using derby to execute quickstart, the table was erroneously accessed before it was created

### Affects Versions

0.4.0

### What engines are you seeing the problem on?

_No response_

### How to reproduce

_No response_

### Relevant log output

POPO20230324-105503

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #1273 

## Brief change log

 prepare derby env before init container/optimize config

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/ams-server/src/main/java/com/netease/arctic/ams/server/ArcticMetaStore.java [hunk 1]
```diff
@@ -186,6 +186,12 @@ private static String getArcticHome() {
   }
 
   public static void startMetaStore(Configuration conf) throws Throwable {
+    //prepare env
+    if (conf.getString(ArcticMetaStoreConf.DB_TYPE).equals("derby")) {
+      DerbyService derbyService = new DerbyService();
+      derbyService.createTable();
+    }
+
```
Comment (baiyangtx) [inline ams/ams-server/src/main/java/com/netease/arctic/ams/server/ArcticMetaStore.java:194]: This may execute createTable at each start. and derby DDL can't be executed for mulit times.

**Review by baiyangtx — CHANGES_REQUESTED** [review]

**Review by baiyangtx — APPROVED** [review]
